### DISCUSSION TITLE
xstate inspect on production?

### DISCUSSION THREAD

**Discussion by ilovett** [statelyai/xstate#1952] [Q&A]
I have xstate inspect check a local storage flag to open in a new tab.  On development environment, the newly opened tab connects properly, but on a production build, the newly opened tab remains in "Waiting for connection..."

Is there some additional configuration required for production?  Is there some automagic setup that works in a development environment but not in a production build ?

**Comment by davidkpiano** [comment]
Peek inside `@xstate/inspect` source code. There is no magic - it's just passing events from window to window. There might be something preventing that in your prod environment (which makes sense).

**Comment by goldo** [comment]
I have the same problem. For context, I'm using nextjs, and when I run in production mode, on vercel, or even on local, it is stuck on "waiting for connection...". When I run in dev mode, it's working fine.

Have you find anything @ilovett  ?

**Reply by goldo** [reply]
Alright, I just found this PR from yesterday 😅 
https://github.com/davidkpiano/xstate/pull/2046/files
